### PR TITLE
Adds the Ninja MODsuit

### LINKED ISSUE #35826: Ninja clothes are sticky
Ninja clothes keep their nodrop on when the ninja explodes and loses them, so anyone who picks it up has the clothing stuck to their hands or wherever its equipped to

### LINKED ISSUE #47980: issue with ninja recalling energy katana
https://www.youtube.com/watch?v=2JfwcBsuBak&feature=youtu.be

basically 1) lights are out. 2)someone else holding energy katana. these are my 2 biggest hypotheses for why recalling katana didnt work. as you can see by me grabbing hypercharged yellow core, i have extremely rapid and self recharging, so it isnt a power issue. I use the adrenal ability, so my suit is certainly initialized and i wasnt stunned.

I've had this issue before, didn't report it the first time because I thought it was just a 1 time fucky wucky, this time confirmed it though

### LINKED ISSUE #67225: Ninja Sword Charges bug out after upgrading
Reporting client version: 514.1583

## Round ID:
[183631](https://scrubby.melonmesa.com/round/183631)

## Testmerges:
- [Resprites almost all the hair to modern standards by Crumpaloo](https://www.github.com/tgstation/tgstation/pull/67184)
- [Adds Cargorilla](https://www.github.com/tgstation/tgstation/pull/67003)
- [Optimizes Runechat](https://www.github.com/tgstation/tgstation/pull/65791)

## Reproduction:
Immediately after upgrading my suit as Ninja the sword seemed to stop recharging its dashes and the icon for it dissapeared completely, when an admin looked into it they said something messed up the timers and it wouldn't charge up past the first one

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

Adds the ninja MODsuit, converting all the ninja abilities into modules and cleaning them up in the meantime.
Removes block tracking and touches from helmets and gloves respectively, use signals for that.
makes the ninja power cell not a varedited high capacity cell
fixes dash beams not working
fixes double anime phobia
adds a new ninja mask and jumpsuit (sprites arent finished yet)
image

fixes #35826
fixes #47980 probably
fixes #67225 probably

- [x] better descriptions for modules/the suit
- [x] sprites for the ninja mask/jumpsuit
- [x] making the ninja glove draining stuff work
- [x] removing the old space suit

## Why It's Good For The Game

Ninja code is pretty bad, I think it's best to move away into nice modular stuff instead.

## Changelog

:cl: Fikou, PositiveEntropy, Nerevar, InfraRedBaron
refactor: the ninja space suit is now a modsuit
fix: fixes dash beams not working
/:cl:

### CODE REVIEW

**Comment by VladinXXV** [comment]
Are we losing any functionality here, or is the plan for it to effectively just be a better ninja suit? The first thing that comes to mind is draining power from wires/apcs/etc, it'd be a big ninja nerf if they had to change their cell or hide in a borg charger any time they got low on power.

**Comment by Fikou** [comment]
> Are we losing any functionality here, or is the plan for it to effectively just be a better ninja suit? The first thing that comes to mind is draining power from wires/apcs/etc, it'd be a big ninja nerf if they had to change their cell or hide in a borg charger any time they got low on power.

it can do anything the ninja suit can

File: code/modules/events/shuttle_loan.dm [hunk 1]
```diff
@@ -155,7 +155,7 @@
 				var/turf/T = pick_n_take(empty_shuttle_turfs)
 
 				new /obj/effect/decal/remains/human(T)
-				new /obj/item/clothing/shoes/space_ninja(T)
+				new /obj/item/clothing/shoes/jackboots/fast(T)
```
Comment (MrMelbert) [inline code/modules/events/shuttle_loan.dm:158]: Thanks I hate it

**Comment by MetalClone** [comment]
Does it still blow up non-ninjas if they try to wear it? Or does it just get deleted when the ninja self destructs?

**Comment by Fikou** [comment]
> Does it still blow up non-ninjas if they try to wear it? Or does it just get deleted when the ninja self destructs?

both

**Comment by MetalClone** [comment]
Fantastic, awesome work again Fikou

File: code/modules/mod/modules/modules_ninja.dm [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+		return
+	mod.wearer.SetUnconscious(0)
+	mod.wearer.SetStun(0)
+	mod.wearer.SetKnockdown(0)
+	mod.wearer.SetImmobilized(0)
+	mod.wearer.SetParalyzed(0)
+	mod.wearer.adjustStaminaLoss(-200)
+	mod.wearer.remove_status_effect(/datum/status_effect/speech/stutter)
+	mod.wearer.reagents.add_reagent(/datum/reagent/medicine/stimulants, 5)
+	mod.wearer.say(pick_list_replacements(NINJA_FILE, "lines"), forced = type)
```
Comment (Cenrus) [inline code/modules/mod/modules/modules_ninja.dm:401]: If you want to make this even more modular, it'd be great if the voice lines only trigger when the suit wearer is a ninja

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/mod/modules/modules_ninja.dm [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+
+/obj/item/mod/module/stealth/proc/on_bullet_act(datum/source, obj/projectile/projectile)
+	SIGNAL_HANDLER
+
+	if(projectile.nodamage)
+		return
+	unstealth(source)
+
+//Advanced Cloaking - Doesn't turf off on bump, less power drain, more stealthy.
+/obj/item/mod/module/stealth/ninja
```
[the remark this answers is not in the document]
Comment (Fikou) [reply code/modules/mod/modules/modules_ninja.dm:65]: no

File: code/modules/mod/modules/modules_ninja.dm [hunk 4]
```diff
@@ -0,0 +1,442 @@
...
+	icon_state = "dnalock_ninja"
+	use_power_cost = DEFAULT_CHARGE_DRAIN * 0.5
+
+/obj/item/mod/module/dna_lock/reinforced/on_mod_activation(datum/source, mob/user)
+	. = ..()
+	if(. != MOD_CANCEL_ACTIVATE || !isliving(user))
+		return
+	var/mob/living/living_user = user
+	to_chat(living_user, span_danger("<B>fATaL EERRoR</B>: 382200-*#00CODE <B>RED</B>\nUNAUTHORIZED USE DETECteD\nCoMMENCING SUB-R0UTIN3 13...\nTERMInATING U-U-USER..."))
+	living_user.gib()
```
[the remark this answers is not in the document]
Comment (Fikou) [reply code/modules/mod/modules/modules_ninja.dm:262]: meh

File: code/modules/mod/mod_control.dm [hunk 5]
```diff
@@ -216,6 +216,7 @@
 			. += span_notice("You could remove [ai] with an <b>intellicard</b>.")
 		else
 			. += span_notice("You could install an AI with an <b>intellicard</b>.")
+	. += span_notice("<i>You could examine it more thoroughly...</i>")
```
Comment (AnturK) [inline code/modules/mod/mod_control.dm:219]: This feels like something that should be automated, but i guess that's for another PR

File: code/modules/mod/mod_control.dm [hunk 6]
```diff
@@ -507,7 +508,7 @@
 		return
 	var/module_reference = display_names[pick]
 	var/obj/item/mod/module/picked_module = locate(module_reference) in modules
-	if(!istype(picked_module) || user.incapacitated())
+	if(!istype(picked_module))
```
Comment (AnturK) [inline code/modules/mod/mod_control.dm:511]: Why the check removal here ?

File: code/game/objects/items.dm [hunk 7]
```diff
@@ -707,29 +707,31 @@ GLOBAL_DATUM_INIT(fire_overlay, /mutable_appearance, mutable_appearance('icons/e
...
-			if (throwforce > 0)
-				if (mob_throw_hit_sound)
-					playsound(hit_atom, mob_throw_hit_sound, volume, TRUE, -1)
-				else if(hitsound)
-					playsound(hit_atom, hitsound, volume, TRUE, -1)
-				else
-					playsound(hit_atom, 'sound/weapons/genhit.ogg',volume, TRUE, -1)
+	if(QDELETED(hit_atom))
+		return
+	if(SEND_SIGNAL(src, COMSIG_MOVABLE_IMPACT, hit_atom, throwingdatum) & COMPONENT_MOVABLE_IMPACT_NEVERMIND)
```
Comment (AnturK) [inline code/game/objects/items.dm:712]: Did you check that current uses of this all return flag values

File: code/modules/ninja/ninjaDrainAct.dm [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+	return NONE
+
+//APC//
+/obj/machinery/power/apc/ninjadrain_act(mob/living/carbon/human/ninja, obj/item/mod/module/hacker/hacking_module)
+	if(!ninja || !hacking_module)
+		return NONE
+	INVOKE_ASYNC(src, .proc/ninjadrain_charge, ninja, hacking_module)
+	return COMPONENT_CANCEL_ATTACK_CHAIN
+
+/obj/machinery/power/apc/proc/ninjadrain_charge(mob/living/carbon/human/ninja, obj/item/mod/module/hacker/hacking_module)
```
Comment (AnturK) [inline code/modules/ninja/ninjaDrainAct.dm:21]: I don't see this one flag value, and they all should.

**Review by AnturK — CHANGES_REQUESTED** [review]

**Pull request merged**